kura/test/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider.test/src/test/java/org/eclipse/kura/cloudconnection/sparkplug/mqtt/endpoint/test/SubscriptionsMapTest.java

```diff
@@ -0,0 +1,196 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Eurotech and/or its affiliates and others
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.cloudconnection.sparkplug.mqtt.endpoint.test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.kura.cloudconnection.sparkplug.mqtt.endpoint.SubscriptionRecord;
+import org.eclipse.kura.cloudconnection.sparkplug.mqtt.endpoint.SubscriptionsMap;
+import org.eclipse.kura.cloudconnection.subscriber.listener.CloudSubscriberListener;
+import org.junit.Test;
+
+public class SubscriptionsMapTest {
+
+    private SubscriptionsMap subscriptionsMap = new SubscriptionsMap();
+    private List<String> returnedTopicsToUnsubscribe = new LinkedList<>();
+    private List<CloudSubscriberListener> returnedMatchingListeners = new LinkedList<>();
+    private CloudSubscriberListener l1 = mock(CloudSubscriberListener.class);
+    private CloudSubscriberListener l2 = mock(CloudSubscriberListener.class);
+    private CloudSubscriberListener l3 = mock(CloudSubscriberListener.class);
+
+    /*
+     * Scenarios
+     */
+
+    @Test
+    public void removeSomeShouldNotReturnTopicsToUnsubscribe() {
+        givenAdd("t1", 0, this.l1);
+        givenAdd("t1", 0, this.l2);
```

Comment:
Here too, I would try to make everything more explicit

```suggestion
        givenSubscriptionMapWith("topicFilter1", 0, this.subListener1);
        givenSubscriptionMapWith("topicFilter1", 0, this.subListener2);
```